lib/xml/xml.go

```diff
@@ -163,10 +163,24 @@ func builtinTypeFor(typ xsd.Type) xsd.Builtin {
 func Unmarshal(r io.Reader, details map[string]Detail) (cdata string, elems map[string]any, err error) {
```

Comment:
I'm not convinced it's necessary — it would be surprising if it did not validate, I expected it to work essentially as std `xml.Decoder` does and was very surprised when it didn't, All up. I'd like to pin down the semantics before we do add any additional docs.

Reply:
Fair point... The doc does not have to explicitly say that it will validate too.